app/controllers/images_controller.rb

```
@@ -0,0 +1,33 @@
+class ImagesController < ApplicationController
+  protect_from_forgery with: :exception
+
+  def create
+    @image = Image.create(image_params)
+
+    if @image.valid?
```

are you sure that `.valid?` will capture all the ways `create` could fail? I'm not saying it won't, I'm just not sure because I've mainly seen `.create!` in the past. The difference between methods with and without a `!` is (usually) that `!` methods will throw an error if they fail; the others will just fail silently, maybe returning false, etc.  So I've usually seen this like 
```
@image = Image.new(image_params)

if @image.valid?
  @image.save!
...
```
Of course, you don't have to do it this way, I just want to make sure we aren't missing any cases, since I'm not as familiar with this approach. If you are comfortable with it, I trust you.

---

That's a good question. I kinda remember getting this from a tutorial, but let me double check on the exact behavior.

---

ok, `create` just calls `save` so if there is an error from e.g. the database, it will not be thrown as an exception. changed it to `create!`